spras/util.py

```diff
@@ -22,7 +34,7 @@ def hash_params_sha1_base32(params_dict: Dict[str, Any], length: Optional[int] =
     @param length: the length of the returned hash, which is ignored if it is None, < 1, or > the full hash length
     """
     params_hash = hashlib.sha1()
-    params_encoded = json.dumps(params_dict, sort_keys=True).encode()
+    params_encoded = json.dumps(params_dict, sort_keys=True, cls=cls).encode()
```

Comment:
We should make a SPRAS release after merging this. I expect the consequence is that the sha1 hashes of parameter combinations will change, breaking output caching. That is, if a user ran PathLinker with k=5 before and they run it again with k=5 after merging this, the results would be regenerated with a new parameter hash.